src/components/needLeadViewPage/needsViewTable.tsx

```diff
@@ -0,0 +1,287 @@
+  });
+
+  useEffect(() => {
+    if (filterStr !== "") {
+      let totalNeeds = filteredNeeds?.getAllNeeds?.needs || [];
+      totalNeeds = totalNeeds
+        .filter((need: any) => {
+          return need.itemName.includes(filterStr);
+        })
+        .slice(0, 10);
+      let space = 10 - totalNeeds.length;
+      for (let i = 0; i < space; i++) {
+        totalNeeds.push({
+          urgencyLevel: "",
+          itemName: "",
+          createdAt: "",
+          specifications: "",
+          quantity: "",
+          budget: "",
+          id: `spacer-need-${Math.random()}`
+        });
+      }
+      setNeeds(totalNeeds);
+      setTotalPage(1);
+    } else {
+      const totalNeeds = filteredNeeds?.getAllNeeds?.needs || [];
+      const needs = needsData?.getAllNeeds?.needs || [];
+      let space;
+      if (needs.length !== 10) {
+        space = 10 - (needs.length % 10);
+      } else {
+        space = 0;
+      }
+
+      for (let i = 0; i < space; i++) {
+        needs.push({
+          urgencyLevel: "",
+          itemName: "",
+          createdAt: "",
+          specifications: "",
+          quantity: "",
+          budget: "",
+          id: `spacer-need-${Math.random()}`
+        });
+      }
+
+      let totalPages = Math.ceil(totalNeeds.length / 10);
+      setNeeds(needs);
+      setTotalPage(totalPages);
+    }
+  }, [needsData, filterStr, totalPages, filteredNeeds]);
+
+  function searchNeeds(e: any) {
+    setFilterStr(e.target.value);
+  }
+
+  const needRow = (need: Need) => (
+    <tr key={need.id}>
+      <td className={styles.td}>{need.urgencyLevel}</td>
+      <td className={styles.td}>{need.itemName}</td>
+      <td className={styles.td}>{need.createdAt}</td>
+      <td className={styles.td}>{need.createdAt}</td>
+      <td className={styles.td}>{need.specifications}</td>
+      <td className={styles.td}>{need.quantity}</td>
+      <td className={styles.td}>{need.createdAt}</td>
+      <td className={styles.td}>{need.budget}</td>
+    </tr>
+  );
+
+  const changePageHandle = (dir: String) => {
+    if (dir === "left" && page > 1) {
+      setPage(page - 1);
+    } else if (dir === "right" && page < totalPages) {
+      setPage(page + 1);
+    }
+  };
+
+  const noDataRow = () => (
+    <tr>
+      <td></td>
+    </tr>
+  );
+
+  return (
+    <>
+      <Box className={styles.boxNeed}>
+        <Box className={styles.needHeader}>
+          <h1 className={styles.headerText}>NEEDS</h1>
+          <Container className={styles.inputBox}>
+            <input
+              className={styles.input}
+              type="text"
+              onChange={searchNeeds}
+            />
+            <SearchIcon className={styles.searchIcon} />
+          </Container>
+        </Box>
+        <Box className={styles.outerTable}>
+          <div className={styles.tableCustomBorder}></div>
+          <table className={styles.table}>
+            <thead>
+              <tr>
+                <th className={styles.header}>
```

Comment:
Optional: You coud define the columns header in an array before, then do a .map(à on it to render the <th> elements